bindata/cloud-network-config-controller/self-hosted/controller.yaml

```diff
@@ -26,7 +26,11 @@ spec:
         openshift.io/component: network
     spec:
       securityContext:
+{{- if not (eq .RunAsUser "")}}
+        runAsUser: {{.RunAsUser}}
+{{- else }}
```

Comment:
According the [API docs](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#securitycontext-v1-core), `If true, the Kubelet will validate the image at runtime to ensure that it does not run as UID 0 (root) and fail to start the container if it does`. 

This is the problem we are running into running HyperShift on the AKS cluster. It's wanting to run the image as root.

Reply:
I get that, but when UID is specified with RunAsUser this should no longer be a problem? I think that something like this should work:
```
 runAsUser: 1001
 runAsNonRoot: true
```

Reply:
I'm happy to give that a go and see. I can always open another PR if it is an issue. I made those changes to `managed/controller.yaml` since you said we didn't need to modify `self-hosted/controller.yaml'.